Content.Shared/Storage/EntitySystems/SharedStorageSystem.cs

```diff
@@ -800,7 +800,6 @@ public void UpdateAppearance(Entity<StorageComponent?, AppearanceComponent?> ent
         _appearance.SetData(uid, StorageVisuals.Capacity, capacity, appearance);
         _appearance.SetData(uid, StorageVisuals.Open, isOpen, appearance);
         _appearance.SetData(uid, SharedBagOpenVisuals.BagState, isOpen ? SharedBagState.Open : SharedBagState.Closed, appearance);
-        _appearance.SetData(uid, StackVisuals.Hide, !isOpen, appearance);
```

Comment:
StackVisuals.Hide is used in YAML for hiding donut box contents and such.

Can you add a field on StorageComponent that makes it not set StackVisuals.Hide instead of a field on ItemCounterComponent?